Mc high ka corrections

ffoucart opened the pull request:
## Proposed changes

Add corrections to high absorption opacity regions needed for implicit Monte-Carlo

### Upgrade instructions

NA

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

Commit:
Correction to opacity for implicit Monte-Carlo

ffoucart commented:
Depends on #5902

ffoucart commented:
Rebased against 5902 and ready for review.

kidder commented on src/Evolution/Particles/MonteCarlo/ImplicitMonteCarloCorrections.tpp:
```diff
@@ -0,0 +1,198 @@
+#include <array>
+
+#include "Evolution/Particles/MonteCarlo/NeutrinoInteractionTable.hpp"
+#include "Evolution/Particles/MonteCarlo/TemplatedLocalFunctions.hpp"
```
make this the first include followed by a blank line

kidder commented on src/Evolution/Particles/MonteCarlo/ImplicitMonteCarloCorrections.tpp:
```diff
@@ -0,0 +1,198 @@
+  // d(radiation_energy_density)/d(fluid_energy_density)
+  // at constant Ye, and
+  // d(neutrino_lepton_number_density)/d(proton_number_density)
+  // at constant T as measured of how 'impactfull' neutrino
```
as a measure of how impactful

kidder reviewed (changes requested):
you can squash the change when rebasing

ffoucart commented:
Latest PR addressed code review requests, but also rebased against #6037 to avoid future conflicts. This PR now depends on #6037 (only for minor changes to the interface of the neutrino table when adding GZs)
